gui/common_qt_functions.py

```diff
@@ -248,74 +318,16 @@ def closeEvent(self, event):
         elif ret == QtWidgets.QMessageBox.Save:
             # Insert all settings to yaml file
             self.exitWindowCleanup()
-            if self.writeConfigFile():
+            self.updateSettingsFromUIChoices()
+            if self.writeConfigFile(self.settings):
```

Comment:
I'm pretty sure this is correct, because `closeEvent` is called with `self` set to the widget, so the widget's `self.settings` would be accessible here.
But I'm writing this here, because it's the one place where I'm not completely sure.